graf3d/eve7/src/REveCaloData.cxx

```diff
@@ -134,22 +134,25 @@ REveCaloData::REveCaloData(const char* n, const char* t):
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Process newly selected cells with given select-record.
-/// Secondary-select status is set.
-/// CellSelectionChanged() is called if needed.
 
 void REveCaloData::ProcessSelection(vCellId_t& sel_cells, UInt_t selectionId, Bool_t multiple)
 {
-   REveSelection* selection = (REveSelection*) ROOT::Experimental::gEve->FindElementById(selectionId);
-
-   std::set<int> secondary_idcs;
-   for (vCellId_i i = sel_cells.begin(); i != sel_cells.end(); ++i)
+   if (fSelector)
    {
-      int id = (i->fSlice << 24) + i->fTower;
-      // printf("set set unique ID (%d, %d) -> [%d]\n",i->fSlice,  i->fTower, id);
-      secondary_idcs.insert(id);
+      fSelector->ProcessSelection(sel_cells, selectionId, multiple);
    }
+   else
+   {
+      REveSelection* selection = (REveSelection*) ROOT::Experimental::gEve->FindElementById(selectionId);
 
-   selection->NewElementPicked(GetElementId(), multiple, true, secondary_idcs);
+      std::set<int> secondary_idcs;
+      for (vCellId_i i = sel_cells.begin(); i != sel_cells.end(); ++i)
```

Comment:
Also here: `for (auto &i : sel_cells)`

Reply:
@linev I have left this code unchanged to make it more explicit.